libs/server/Auth/AuthenticationSettings.cs

```diff
@@ -153,13 +158,27 @@ public AadAuthenticationSettings(string[] authorizedAppIds, string[] audiences,
             _signingTokenProvider = signingTokenProvier;
         }
 
+        public AadAuthenticationSettings WithUsernameValidation()
+        {
+            _validateUsername = true;
```

Comment:
I'm trying to avoid having multiple configuration settings in the constructor and allowing flexibility incase we add more settings in future so preferring the Builder pattern here. Also modifying constructor breaks binray compatibility since it's a public class, I'm not sure if someone had a dependency on it

Reply:
It should be fine if you add it with default value. So, you won't have to change it everywhere. I personally feel it is much easier to reason about.